### PR TITLE
chore: Initialize the project

### PR DESCRIPTION
**Pull request by LingKa28**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Initialize the project

* what changes does this pull request make?
  * Initialize the project
  * Add CI 

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
No

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# SPDX-FileCopyrightText: 2023-present LingKa <cnfty786@gmail.com>
+#
+# SPDX-License-Identifier: MIT
```
Comment (bsbds) [inline tests/__init__.py:3]: Same as above.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+dependencies = []
+
+[project.urls]
+Documentation = "https://github.com/unknown/py-xline#readme"
+Issues = "https://github.com/unknown/py-xline/issues"
+Source = "https://github.com/unknown/py-xline"
```
Comment (bsbds) [inline pyproject.toml:32]: Please fix the project url here

**Comment by iGxnon** [comment]
LGTM

File: .gitignore [hunk 3]
```diff
@@ -158,3 +158,8 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+# Custom
+api
+playground
+.vscode
```
[the remark this answers is not in the document]
Comment (LingKa28) [reply .gitignore:165]: Thank you for your feedback. I've updated the pull request based on your suggestions.

File: client/__init__.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# SPDX-FileCopyrightText: 2023-present LingKa <cnfty786@gmail.com>
+#
+# SPDX-License-Identifier: MIT
```
[the remark this answers is not in the document]
Comment (LingKa28) [reply client/__init__.py:3]: This file allows Python to treat the directory containing it as if it were a Python package, and we can do some initialization and so on.

[the remark this answers is not in the document]
Comment (LingKa28) [reply client/__init__.py:3]: This code was generated when I initialized the project with hatch. I forgot to change it. I will modify the license.

File: tests/__init__.py (hunk 1 above)
Comment (LingKa28) [reply tests/__init__.py:3]: Same as above.

File: pyproject.toml (hunk 2 above)
Comment (LingKa28) [reply pyproject.toml:32]: Oops, sorry I missed that! I've updated it as you recommended.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "py-xline"
+dynamic = ["version"]
+description = 'py-xline is an official xline client sdk, written in Python.'
+readme = "README.md"
+requires-python = ">=3.7"
+license = "MIT"
```
Comment (iGxnon) [inline pyproject.toml:11]: MIT ?

**Review by iGxnon — APPROVED** [review]

**Review by bsbds — APPROVED** [review]

**Pull request merged**